src/test/kotlin/org/opensearch/commons/alerting/model/ClusterMetricsInputTests.kt

```diff
@@ -437,6 +437,6 @@ class ClusterMetricsInputTests {
         // THEN
         assertEquals(path, clusterMetricsInput.path)
         assertEquals(pathParams, clusterMetricsInput.pathParams)
-        assertEquals(testUrl, clusterMetricsInput.url)
+        assertEquals(testUrl.replace(",", "%2C"), clusterMetricsInput.url)
```

Comment:
I suppose it's fine for a test, but you're URL-encoding/decoding here, so you should be calling url decode.

Reply:
@dblock fixed the logic so tests no longer need changes. kindly check my comment below.